src/VbsEnclaveSDK/src/veil_host_lib/hello_winrt_kcm.vtl0.cpp

```diff
@@ -0,0 +1,340 @@
+        default:
+            cacheOption = KeyCredentialCacheOption::NoCache; // Default fallback
+            break;
+    }
+
+    // Convert timeout from seconds to TimeSpan (100-nanosecond units)
+    winrt::Windows::Foundation::TimeSpan timeout{ static_cast<int64_t>(cacheConfig.cacheTimeoutInSeconds) * 10000000LL };
+
+    // Use WinRT's get_activation_factory for a more modern approach
+    auto factory = winrt::get_activation_factory<winrt::Windows::Security::Credentials::KeyCredentialCacheConfiguration, 
+                                                 winrt::Windows::Security::Credentials::IKeyCredentialCacheConfigurationFactory>();
+
+    return factory.CreateInstance(cacheOption, timeout, cacheConfig.cacheUsageCount);
+}
+
+veil_abi::Types::credentialAndSessionInfo veil_abi::Untrusted::Implementation::userboundkey_establish_session_for_create(
+    uintptr_t enclave,
+    const std::wstring& key_name,
+    uintptr_t ecdh_protocol,
+    const std::wstring& message,
+    uintptr_t window_id,
+    const veil_abi::Types::keyCredentialCacheConfig& cache_config,
+    uint32_t key_credential_creation_option)
+{
+    std::wcout << L"Inside userboundkey_establish_session_for_create_callback"<< std::endl;
+    auto algorithm = GetAlgorithm(ecdh_protocol);
+
+    // Convert the cacheConfig parameter to KeyCredentialCacheConfiguration
+    auto cacheConfiguration = ConvertCacheConfig(cache_config);
+
+    auto sessionInfo = std::make_shared<uintptr_t>(0);
+    auto enclaveptr = (void*)enclave;   
+
+    try
+    {
+        auto op = KeyCredentialManager::DeleteAsync(key_name);
+        std::wcout << "Deletion worked" << std::endl;
+
+        op.get();
+    }
+    catch (...)
+    {
+        std::wcout << "Deletion failed" << std::endl;
+    }
+    
+    std::wcout << L"Calling RequestCreateAsync" << std::endl;
+    auto credentialResult = KeyCredentialManager::RequestCreateAsync(
+        key_name,
+        static_cast<KeyCredentialCreationOption>(key_credential_creation_option),
+        algorithm,
+        message,
+        cacheConfiguration,
+        static_cast<winrt::Windows::UI::WindowId>(window_id),
+        ChallengeResponseKind::VirtualizationBasedSecurityEnclave,
+        CreateChallengeCallback(sessionInfo, enclaveptr, L"Challenge")
+    ).get();
+
+    std::wcout << L"RequestCreateAsync returned" << std::endl;
+
+    // Check if the operation was successful
+    auto status = credentialResult.Status();
+    if (status != KeyCredentialStatus::Success)
+    {        
+        // Throw after setting result so VTL1 can clean up
+        THROW_HR(static_cast<HRESULT>(status));
+    }
+
+    const auto& credential = credentialResult.Credential();
+
+    // AddRef the COM object to keep it alive when we pass the pointer to VTL1
+    // VTL1 will consume this reference when it's done
+    auto credentialPtr = reinterpret_cast<uintptr_t>(winrt::get_abi(credential));
+    static_cast<IUnknown*>(winrt::get_abi(credential))->AddRef();
```

Comment:
nit: This is logically good, but is there a way to write it slightly more clearly/idiomatically?

(AddRef feels chunky)

```
    wil::com_ptr iunk { winrt::get_abi(credential) };
    result.credential = iunk.release();
```